### PR TITLE
store: findSegments with an empty prevLinkHash

### LINKED ISSUE #181: store: make it possible to findSegments with an empty prevLinkHash
Right now it is not possible to explicitly target segments that have no parents.

### PR DESCRIPTION
**Pull request by alexppxela**
Resolves stratumn/sdk#181

### CODE REVIEW

**Comment by such** [comment]
Do we have to add a new filter param? Is it not possible to make the difference between when PrevLinkHash is absent and when it is empty?

File: store/store.go [hunk 1]
```diff
@@ -106,6 +106,9 @@ type SegmentFilter struct {
 	Process string `json:"process"`
 
 	// A previous link hash the segments must have.
```
Comment (t-bast) [inline store/store.go]: nit: that comment needs to be updated

File: store/store.go [hunk 2]
```diff
@@ -106,6 +106,9 @@ type SegmentFilter struct {
...
+	EmptyPrevLinkHash bool `json:"emptyPrevLinkHash"`
+
+	// A previous link hash the segments must have.
 	PrevLinkHash *types.Bytes32 `json:"prevLinkHash"`
```
Comment (t-bast) [inline store/store.go]: noobie question: why do we add a bool instead of allowing a nil PrevLinkHash and treating that as if EmptyPrevLinkHash was true ?

Comment (t-bast) [reply store/store.go]: Is it because we want to allow searching for both a specific previous link hash and empty link hash in the same request ?

**Comment by alexppxela** [comment]
I do not think that it is possible to determine an empty [32]byte.
Now, it is a pointer to determine if the PrevLinkHash has to be checked during match. I think the only one solution to determine if Match function has to check an empty PrevLinkHash is to had a bool

File: store/store.go (hunk 2 above)
Comment (alexppxela) [reply store/store.go]: nil value is to do not filter segment with a PrevLinkHash. All filter elements are optional

File: store/store.go [hunk 3]
```diff
@@ -106,7 +106,9 @@ type SegmentFilter struct {
 	Process string `json:"process"`
 
 	// A previous link hash the segments must have.
-	PrevLinkHash *types.Bytes32 `json:"prevLinkHash"`
+	// nil makes this attribute as optional
+	// empty string is to search Segment without parent
```
Comment (such) [inline store/store.go]: typo: Segments

Comment (alexppxela) [reply store/store.go]: oops, done.

**Pull request merged**